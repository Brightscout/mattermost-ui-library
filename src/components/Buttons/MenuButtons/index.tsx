import React from 'react';

type MenuButtonProps = {
    buttons: {
        text: string,
        icon: string | JSX.Element;
        onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
    }[];
    className?: string;
}

const MenuButtons = ({buttons, className = ''}: MenuButtonProps): JSX.Element => (
    <div className={`button-menu d-flex flex-column ${className}`}>
        {buttons.map((button) => (
            <button
                key={button.text}
                className='button-menu__btn channel-bg channel-text font-14 padding-v-12 padding-h-10 d-flex plugin-btn'
                onClick={button.onClick}
            >
                <span className='button-menu__btn-icon margin-right-10 d-inline-block'>
                    {typeof button.icon === 'string' ? <i className={button.icon} /> : button.icon}
                </span>
                <span className='button-menu__btn-text'>{button.text}</span>
            </button>
        ))}
    </div>
);

export default MenuButtons;
