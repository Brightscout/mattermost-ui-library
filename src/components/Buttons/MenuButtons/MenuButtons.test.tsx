import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';

import MenuButtons from '.';

// Mock functions
const onBtnClick = jest.fn();

// Mock data for props
const Icon = <i className='icon icon-globe' />
const buttons = [{
    text: 'Button Text',
    icon: Icon,
    onClick: onBtnClick,
}, {
    text: 'Button Text 2',
    icon: Icon,
    onClick: onBtnClick,
}];

describe('MenuButtons', () => {
    let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>

    beforeEach(() => {
        component = shallow(<MenuButtons buttons={buttons} className='menu-btn' />);
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should render the passed icon from the props', () => {
        expect(component.containsMatchingElement(Icon)).toBeTruthy();
    });

    it('Should render the passed button text from the props', () => {
        expect(component.text().includes('Button Text')).toBeTruthy();
    });

    it('Should apply the passed className prop', () => {
        expect(component.hasClass('menu-btn')).toBeTruthy();
    })

    it('Should render all the buttons passed as props', () => {
        expect(component.find('.menu-btn .button-menu__btn')).toHaveLength(2);
    })
})
