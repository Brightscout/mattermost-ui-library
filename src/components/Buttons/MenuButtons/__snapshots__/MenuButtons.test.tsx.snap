// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButtons Should render correctly 1`] = `
<div
  className="button-menu d-flex flex-column menu-btn"
>
  <button
    className="button-menu__btn channel-bg channel-text font-14 padding-v-12 padding-h-10 d-flex plugin-btn"
    key="Button Text"
    onClick={[MockFunction]}
  >
    <span
      className="button-menu__btn-icon margin-right-10 d-inline-block"
    >
      <i
        className="icon icon-globe"
      />
    </span>
    <span
      className="button-menu__btn-text"
    >
      Button Text
    </span>
  </button>
  <button
    className="button-menu__btn channel-bg channel-text font-14 padding-v-12 padding-h-10 d-flex plugin-btn"
    key="Button Text 2"
    onClick={[MockFunction]}
  >
    <span
      className="button-menu__btn-icon margin-right-10 d-inline-block"
    >
      <i
        className="icon icon-globe"
      />
    </span>
    <span
      className="button-menu__btn-text"
    >
      Button Text 2
    </span>
  </button>
</div>
`;
